### ISSUE TITLE
insert_function_instance resource does not work with ID output from segment_function resource for insert functions

### ISSUE THREAD

**Issue by backjo** [segmentio/terraform-provider-segment#85]
**Description**
The segment_insert_function_instance resource does not work with the ID being output from the segment_function resource as show in the documents. This appears to be due to the ifnd_ prefix being output from segment_function, which causes a 404 not found on the insert-function-instances API because it expects the prefix to be trimmed / not be present

**To Reproduce**

Example Terraform code that triggers the issue:

```terraform
resource "segment_function" "amplitude_insert_function" {
  code          = <<EOF
EOF
  display_name  = "Amplitude Insert Function"
  logo_url      = "https://placekitten.com/200/138"
  resource_type = "INSERT_DESTINATION"
  description   = "This is function to insert context into amplitude events"
  settings      = []
}
resource "segment_insert_function_instance" "amplitude_instance" {
  integration_id = segment_destination.my_amplitude_instance.id
  function_id    = segment_function.amplitude_insert_function.id
  name           = "Amplitude Destination Function"
  enabled        = true
  settings       = jsonencode({})
}
```

**Comment by cv** [comment]
Thank you for the report, @backjo! You're right, the prefix needs to be stripped from the ID. You can use the built-in https://developer.hashicorp.com/terraform/language/functions/trimprefix for this.

We'll update the documentation shortly to reflect that.

**Comment by backjo** [comment]
hey @cv   - that's totally what we're doing to workaround. Ideally, it'd behave like the other TF resources and I'd be able to take the output ID from the insert_function as the argument for function_id

**Comment by deanhuynh** [comment]
Hi @backjo this has been fixed in v0.10.0. Thanks for waiting!
